qt/aqt/editor.py

```diff
@@ -579,8 +581,12 @@ def _update_duplicate_display(self, result: NoteFieldsCheckResult.V) -> None:
         elif result == NoteFieldsCheckResult.FIELD_NOT_CLOZE:
             cloze_hint = tr.adding_cloze_outside_cloze_field()
 
-        self.web.eval(f"uiPromise.then(() => setBackgrounds({json.dumps(cols)}));")
-        self.web.eval(f"uiPromise.then(() => setClozeHint({json.dumps(cloze_hint)}));")
+        self.web.eval(
+            'require("anki/ui").uiDidLoad.then(() => {'
+            f"setBackgrounds({json.dumps(cols)});\n"
```

Comment:
I guess the use of globals here is legacy - presumably in the future we'd have setBackgrounds() or similar exported in a module like anki/editor?

Reply:
I think most of the `set...` functions like `setFields` and `setSticky` could by now be replaced with calls directly to the backend. The editor could call `NoteFieldsCheck` directly and set the backgrounds accordingly.